packages/core/src/utils/indyIdentifiers.ts

```
@@ -0,0 +1,114 @@
+export type IndyNamespace = `${Did}:${DidIndyMethod}:${DidIndyNamespace}:${NamespaceIdentifier}`
+
+export function isQualifiedIndyIdentifier(identifier: string | undefined): boolean {
+  if (!identifier || identifier === '') return false
+  return identifier.startsWith('did:indy:')
+}
+
+export function getLegacyIndyCredentialDefinitionId(qualifiedIdentifier: string) {
+  if (!isQualifiedIndyIdentifier(qualifiedIdentifier)) return qualifiedIdentifier
+
+  const lastColonIndex = qualifiedIdentifier.lastIndexOf(':')
+  const identifierTrunk = qualifiedIdentifier.substring(lastColonIndex + 1)
+  const txType = identifierTrunk.split('/')[3]
+
+  if (txType === 'CLAIM_DEF') {
+    // did:indy:sovrin:5nDyJVP1NrcPAttP3xwMB9/anoncreds/v0/CLAIM_DEF/56495/npdb -> 5nDyJVP1NrcPAttP3xwMB9:3:CL:56495:npbd
+    const [id, , , , seqNo, name] = identifierTrunk.split('/')
+    return generateCredentialDefinitionId(id, +seqNo, name)
+  } else {
+    // indy Union Indy Ledger w/o url syntax
+    // did:indy:idunion:test:2MZYuPv2Km7Q1eD4GCsSb6 -> 2MZYuPv2Km7Q1eD4GCsSb6
+    return identifierTrunk
+  }
+}
+
+export function getLegacyIndySchemaId(qualifiedIdentifier: string) {
+  if (!isQualifiedIndyIdentifier(qualifiedIdentifier)) return qualifiedIdentifier
+
+  const lastColonIndex = qualifiedIdentifier.lastIndexOf(':')
+  const identifierTrunk = qualifiedIdentifier.substring(lastColonIndex + 1)
+  const txType = identifierTrunk.split('/')[3]
+
+  if (txType === 'SCHEMA') {
+    // did:indy:sovrin:F72i3Y3Q4i466efjYJYCHM/anoncreds/v0/SCHEMA/npdb/4.3.4 -> F72i3Y3Q4i466efjYJYCHM:2:npdb:4.3.4
+    const [id, , , , name, version] = identifierTrunk.split('/')
+    return generateSchemaId(id, name, version)
+  } else {
+    // indy Union Indy Ledger w/o url syntax
+    // did:indy:idunion:test:2MZYuPv2Km7Q1eD4GCsSb6 -> 2MZYuPv2Km7Q1eD4GCsSb6
+    return identifierTrunk
+  }
```

Same here, would be better to throw an error in this case I think